modules/core/shared/src/main/scala-3/io/circe/derivation/package.scala

```diff
@@ -25,28 +25,58 @@ private[circe] inline final def summonLabels[T <: Tuple]: List[String] =
     case _: EmptyTuple => Nil
     case _: (t *: ts)  => constValue[t].asInstanceOf[String] :: summonLabels[ts]
 
-private[circe] inline final def summonEncoders[T <: Tuple](using Configuration): List[Encoder[_]] =
+private[circe] inline final def summonEncoders[T <: Tuple](inline autoRecurse: Boolean)(using
+  Configuration
+): List[Encoder[_]] =
```

Comment:
I have reservations about changing the signature of any package private methods, we've had them break bincompat in surprising ways that mima refuses to verify for us.  Can you find all the package private ones you've changed and add deprecated versions that call your new one? 

Once you've gotten through that, @zmccoy and I will look over it.

Reply:
> we've had them break bincompat in surprising ways that mima refuses to verify for us

Yes, probably those methods should be marked with the `@binaryAPI` annotation currently going through SIP.

- https://github.com/scala/improvement-proposals/pull/58

Reply:
@zarthross addressed!